inspirehep/modules/workflows/utils/__init__.py

```diff
@@ -153,14 +153,35 @@ def _try_to_log(logfn, *args, **kwargs):
     return _decorator
 
 
+@contextmanager
 @with_debug_logging
-def get_pdf_in_workflow(obj):
-    """Return the fullpath to the PDF attached to a workflow object"""
-    for filename in obj.files.keys:
-        if filename.endswith('.pdf'):
-            return retrieve_uri(obj.files[filename].file.uri)
+def get_document_in_workflow(obj):
+    """Context manager giving the path to the document attached to a workflow object.
 
-    obj.log.info('No PDF available')
+    Arg:
+        obj: workflow object
+
+    Returns:
+        Optional[str]: The path to a local copy of the document.  If no
+        documents are present, it retuns None.  If several documents are
+        present, it prioritizes the fulltext. If several documents with the
+        same priority are present, it takes the first one and logs an error.
+    """
+    documents = obj.data.get('documents', [])
+    fulltexts = [document for document in documents if document.get('fulltext')]
+    documents = fulltexts or documents
+
+    if not documents:
+        obj.log.info('No document available')
+        yield None
```

Comment:
Just fyi, the `None` is the default if nothing passed.

Reply:
I know, but PEP8 says
>Be consistent in return statements. Either all return statements in a function should return an expression, or none of them should. If any return statement returns an expression, any return statements where no value is returned should explicitly state this as return None, and an explicit return statement should be present at the end of the function (if reachable).

Reply:
then you might be missing a `None` in the return right after this no?